### DISCUSSION TITLE
How to update list item

### DISCUSSION THREAD

**Discussion by Clashkid155** [charmbracelet/bubbles#357] [Q&A]
From my previous question #352, I want to update the value of a  list item.
My current solution is to update the item and then reinsert it at the index of the previous value.

```go
type listItem struct {
	name      string
	triggered bool
}

case tea.KeyMsg:
		switch msg.String() {
		case "enter":
			lis := m.list.SelectedItem().(listItem)
			lis.triggered = !lis.triggered
			m.list.SetItem(m.list.Index(), lis)
		}
``` 
I want to know if this is an ideal solution or if there's a better way.

**Comment by naglis** [comment]
I am not sure if there is a better way to update a list item (I am using the same approach), but please note that `SelectedItem()` [may return nil](https://github.com/charmbracelet/bubbles/blob/95d7be54d9d4cdd73cd36e31e64b67965d517d68/list/list.go#L442) in case e.g. the list is empty, so an extra not-nil check should be done before the type assertion and `list.SetItem` [returns a command](https://github.com/charmbracelet/bubbles/blob/95d7be54d9d4cdd73cd36e31e64b67965d517d68/list/list.go#L349), so it should be handled to correctly update the state of the list e.g. in case of an active filter.
